olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Messages.h

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <olp/dataservice/read/DataServiceReadApi.h>
+#include <olp/dataservice/read/model/Data.h>
+#include <olp/dataservice/read/model/StreamOffsets.h>
+#include <boost/optional.hpp>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+namespace model {
+
+/**
+ * @brief Class, which encapsulates info about actual data content.
+ */
+class DATASERVICE_READ_API Metadata final {
+ public:
+  /**
+   * @return Returns the
+   * [partition](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/layers/partitions.html)
+   * which this metadata content is related to.
+   */
+  const std::string& GetPartition() const { return partition_; }
+  /**
+   * @brief Specifies partition of this metadata content.
+   *
+   * @param value The partition id string. Maximum length is 500 characters.
+   */
+  void SetPartition(std::string value) { partition_ = std::move(value); }
+
+  /**
+   * @return The checksum of the content.
+   */
+  const boost::optional<std::string>& GetChecksum() const { return checksum_; }
+  /**
+   * @brief Specified checksum of this metadata content.
+   *
+   * The algorithm used to calculate the checksum
+   * is user specific. Algorithms that can be used are, for example, MD5 or
+   * SHA1. This is not a secure hash, it's used only to detect changes in
+   * content.
+   * @param value Checksum string.
+   */
+  void SetChecksum(std::string value) { checksum_ = std::move(value); }
```

Comment:
Usually if the member is of type ```boost::optional``` then also the setter will take in a ```boost::optional``` as input.